extensions/wikia/FilePage/WikiaFilePage.php

```diff
@@ -40,29 +40,96 @@ protected function isVideo() {
 	/**
 	 * Render the image or video
 	 */
-	protected function openShowImage(){
+	protected function openShowImage() {
 		if ( $this->isVideo() ) {
 			$this->openShowVideo();
 		} else {
 			parent::openShowImage();
 		}
 	}
 
+	/**
+	 * Render the image or video
+	 */
+	public function view() {
+		global $wgMemc, $wgDBprefix, $wgOut, $wgUser, $wgGroupPermissionsLocal, $wgNamespaceProtection;
+		if ( !$wgUser->isAnon() ) {
+			parent::view();
+
+			return;
+		}
+		//fallback to main page
+		$url = wfExpandUrl( "/wiki/Main_Page" );
+		//wiki needs read privileges
+		if ( preg_match( "/\*\|read\|0/", $wgGroupPermissionsLocal ) ||
+			 preg_match( "/\user\|read\|0/", $wgGroupPermissionsLocal ) ) {
+			$wgOut->redirect( $url );
+
+			return;
+		}
+		$redirKey = "redir:" . $wgDBprefix . ":" . $this->getTitle();
+		$url = $wgMemc->get( $redirKey );
+		if ( $url ) {
+			$wgOut->redirect( $url );
+		}
+		$displayImg = $img = false;
+		Hooks::run( 'ImagePageFindFile', [ $this, &$img, &$displayImg ] );
+		if ( !$img ) { // not set by hook?
+			$img = wfFindFile( $this->getTitle() );
+			if ( !$img ) {
+				$img = wfLocalFile( $this->getTitle() );
+			}
+		}
+		if ( !$img ) {
+			parent::view();
+
+			return;
+		}
+		$res = $this->fetchLinks( $img->getTitle()->getDBkey() );
+
+		foreach ( $res as $row ) {
+			$title = Title::newFromRow( $row );
+			if ( $title->isRedirect() ) {
+				continue;
+			}
+			$url = wfExpandUrl( "/wiki/" . $row->page_title );
```

Comment:
`$title->getFullURL()` should be used to get hte URL from the title. Hardcoding things like `/wiki/` will break situations like language path wikis, or if we ever changed `wgArticlePath`, so using the methods to build the URL will account for all that for you.